hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -2589,29 +2588,7 @@ protected Expression consumeSqmParameter(SqmParameter sqmParameter) {
 
 		assert parameterSqmType != null;
 
-		if ( parameterSqmType instanceof SqmPath ) {
-			final SqmPath sqmPath = (SqmPath) parameterSqmType;
-			final NavigablePath navigablePath = sqmPath.getNavigablePath();
-			if ( navigablePath.getParent() != null ) {
-				final TableGroup tableGroup = getFromClauseAccess().getTableGroup( navigablePath.getParent() );
-				return tableGroup.getModelPart().findSubPart(
-						navigablePath.getLocalName(),
-						null
-				);
-			}
-
-			return getFromClauseAccess().getTableGroup( navigablePath ).getModelPart();
-		}
-
-		if ( parameterSqmType instanceof BasicValuedMapping ) {
```

Comment:
This case is not handled in `org.hibernate.metamodel.model.domain.internal.MappingMetamodelImpl#resolveMappingExpressable` and should probably replace the `BasicType` case.

Reply:
Great catch!! I enriched the `MappingMetamodelImpl#resolveMappingExpressable` to accept BasicValuedMapping as well (or any other MappingModelExpressable) as following:
```
if ( sqmExpressable instanceof MappingModelExpressable ) {
	return (MappingModelExpressable) sqmExpressable;
}
```
it also make the following statement unnecessary so I deleted it:
```
if ( sqmExpressable instanceof BasicType<?> ) {
	return (BasicType) sqmExpressable;
}
```